### PR TITLE
Changes in ci and dockerfile

### PR DESCRIPTION
**Pull request by Smolevich**
- Use official image composer in Dockerfile
- Stop containers after in end `after_script`
- Add caching vendor dir

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
-RUN cd /code && composer install --prefer-source --no-interaction
+COPY --from=0 /usr/bin/composer /usr/local/bin/composer
 
 WORKDIR /code
-ENV PATH="~/.composer/vendor/bin:./vendor/bin:${PATH}"
+
+ADD composer.json composer.json
+
+# composer.lock isn't exists in git repository
+#ADD composer.lock composer.lock
```
Comment (jenssegers) [inline Dockerfile]: You can remove these lines :)

Comment (Smolevich) [reply Dockerfile]: Ok

Comment (Smolevich) [reply Dockerfile]: Please, in free time review this PR https://github.com/jenssegers/laravel-mongodb/pull/1614 ;)

File: .travis.yml [hunk 2]
```diff
@@ -4,19 +4,25 @@ language: php
...
   - docker-compose build --build-arg PHP_VERSION=${TRAVIS_PHP_VERSION}
 
 script:
-  - docker-compose run tests composer install --prefer-source --no-interaction && php ./vendor/bin/phpunit --coverage-clover ./clover.xml
+  - docker-compose run tests
 
 after_script:
   - docker-compose run tests php ./vendor/bin/php-coveralls -v
+  - docker-compose ps
+  - docker-compose stop
```
Comment (jenssegers) [inline .travis.yml:28]: Do we need this? Probably this adds some additional time to the build while Travis does it afterwards anyways.

File: Dockerfile [hunk 3]
```diff
@@ -1,19 +1,26 @@
 ARG PHP_VERSION=7.2
+ARG COMPOSER_VERSION=1.8
 
+FROM composer:${COMPOSER_VERSION}
```
Comment (jenssegers) [inline Dockerfile:4]: Nice find!

File: Dockerfile [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
-
-RUN curl -sS https://getcomposer.org/installer | php \
-    && mv composer.phar /usr/local/bin/ \
-    && ln -s /usr/local/bin/composer.phar /usr/local/bin/composer
+    pecl install mongodb && docker-php-ext-enable mongodb
 
-ADD composer.json /code/composer.json
-ADD composer.lock /code/composer.lock
-RUN cd /code && composer install --prefer-source --no-interaction
+COPY --from=0 /usr/bin/composer /usr/local/bin/composer
```
Comment (jenssegers) [inline Dockerfile:20]: Can be `--from=composer`

File: Dockerfile [hunk 5]
```diff
@@ -1,19 +1,26 @@
...
 
 WORKDIR /code
-ENV PATH="~/.composer/vendor/bin:./vendor/bin:${PATH}"
+
+ADD composer.json composer.json
+
+RUN composer global require hirak/prestissimo
```
Comment (jenssegers) [inline Dockerfile:26]: Would skip this.

File: Dockerfile [hunk 6]
```diff
@@ -1,19 +1,26 @@
...
 
-ADD composer.json /code/composer.json
-ADD composer.lock /code/composer.lock
-RUN cd /code && composer install --prefer-source --no-interaction
+COPY --from=0 /usr/bin/composer /usr/local/bin/composer
 
 WORKDIR /code
-ENV PATH="~/.composer/vendor/bin:./vendor/bin:${PATH}"
+
+ADD composer.json composer.json
```
Comment (jenssegers) [inline Dockerfile:24]: Shouldn't we copy the code as well?

Comment (jenssegers) [reply Dockerfile:24]: Or just have no code copied at all and rely on the volume.

File: Dockerfile (hunk 4 above)
Comment (Smolevich) [reply Dockerfile:20]: Your variant can be simple for understanding

**Comment by Smolevich** [comment]
@jenssegers, thanks for review!

**Pull request merged**